examples/BuddyWhisper/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+add_custom_command(
+  OUTPUT ${BUDDY_EXAMPLES_DIR}/BuddyWhisper/forward.mlir ${BUDDY_EXAMPLES_DIR}/BuddyWhisper/subgraph0.mlir ${BUDDY_EXAMPLES_DIR}/BuddyWhisper/arg0.data
+  COMMAND ${Python3_EXECUTABLE} ${BUDDY_EXAMPLES_DIR}/BuddyWhisper/import-whisper.py
+  COMMENT "Generating forward.mlir, subgraph0.mlir and arg0.data..."
+)
+set(PATTERN_ARG "test-generalize-pad-tensor")
+add_custom_command(
+  OUTPUT forward.o
+  COMMAND ${LLVM_MLIR_BINARY_DIR}/mlir-opt ${BUDDY_EXAMPLES_DIR}/BuddyWhisper/forward.mlir 
+            -pass-pipeline "builtin.module(func.func(tosa-to-linalg-named, tosa-to-linalg, tosa-to-tensor, tosa-to-arith), empty-tensor-to-alloc-tensor, convert-elementwise-to-linalg, arith-bufferize, func.func(linalg-bufferize, tensor-bufferize), func-bufferize)" |
+          ${BUDDY_BINARY_DIR}/buddy-opt
+            -pass-pipeline "builtin.module( func.func(buffer-deallocation-simplification, convert-linalg-to-loops),matmul-paralell-vectorization-optimize, batchmatmul-optimize, eliminate-empty-tensors,func-bufferize-dynamic-offset, func.func(llvm-request-c-wrappers),convert-scf-to-openmp, convert-openmp-to-llvm, convert-math-to-llvm, convert-math-to-libm, convert-scf-to-cf,  convert-arith-to-llvm, expand-strided-metadata, finalize-memref-to-llvm, convert-func-to-llvm, reconcile-unrealized-casts)" |
+          ${LLVM_MLIR_BINARY_DIR}/mlir-translate -mlir-to-llvmir |
+          ${LLVM_MLIR_BINARY_DIR}/llvm-as |
+          ${LLVM_MLIR_BINARY_DIR}/llc -filetype=obj  -relocation-model=pic -O0 -o ${BUDDY_BINARY_DIR}/../examples/BuddyWhisper/forward.o
+  DEPENDS ${BUDDY_EXAMPLES_DIR}/BuddyWhisper/forward.mlir
+  COMMENT "Building forward.o"
+  VERBATIM)
+
+add_custom_command(
+    OUTPUT subgraph0.o
+    COMMAND ${LLVM_MLIR_BINARY_DIR}/mlir-opt ${BUDDY_EXAMPLES_DIR}/BuddyWhisper/subgraph0.mlir 
+    -pass-pipeline "builtin.module(func.func(tosa-to-linalg-named, tosa-to-linalg, tosa-to-tensor, tosa-to-arith))" |
+    ${LLVM_MLIR_BINARY_DIR}/mlir-opt
+    -test-linalg-transform-patterns=${PATTERN_ARG} |
+    ${BUDDY_BINARY_DIR}/buddy-opt
+            -arith-expand
+            -eliminate-empty-tensors
+            -convert-elementwise-to-linalg
+            -empty-tensor-to-alloc-tensor
+            -one-shot-bufferize
```

Comment:
Is there some problem with the pass here, why use -one-shot-bufferize, -tensor-bufferize, -arith-bufferize and -func-bufferize at the same time.

Reply:
Thanks for your reply. The ‘func-bufferize’ and 'arith-bufferize' is unnecessary here, and I will delete it.  But after the necessary pass 'func-bufferize-dynamic-offset'(able to achieve the same functions as ‘func-bufferize’),  'bufferization.to_memref' will convert to 'bufferization.to_tensor', so 'tensor-bufferize' is necessary here.

Reply:
Maybe you could try moving `-one-shot-bufferize` behind `func-bufferize-dynamic-offset`.

Reply:
I just tried this, but still encountered some errors, some op generated by `func-bufferize-dynamic-offset` are not supported by One-Shot Analysis, so I need add `tensor-bufferize` after `unc-bufferize-dynamic-offset`.

Reply:
I looked at the code of `func-bufferize-dynamic-offset`, but there is no test case. I am not very clear about what this pass does. It integrates `func-bufferize` and `returnOp-bufferize`. However, I think this is a good opportunity to implement `buddy-one-shot-bufferize`. `buddy-one-shot-bufferize` integrates our own bufferize on the `one-shot-bufferize` basis.
@zhanghb97 What do you think? @qingqing12138 Do you want to make it happen? Maybe I can give you some help.

Reply:
`func-bufferize-dynamic-offset` is used to offset `memref`. Since we have to slice the entire parameter file, offset is needed to find the locations of different parameter objects.
We can indeed make some improvements on `one-shot-bufferize`. I think it is a nice proposal.

Reply:
After you have submitted the code based on the reviewer's suggestion, there should be a click on the button Resolve conversation.This enables the reviewer to guide what changes have been made.